mcl_signer/cmd/signer/mcl_signer.go

```diff
@@ -0,0 +1,49 @@
+package main
+
+import (
+	"encoding/hex"
+	"fmt"
+	"os"
+
+	"github.com/ElrondNetwork/elrond-go/crypto/signing"
+	"github.com/ElrondNetwork/elrond-go/crypto/signing/mcl"
+	"github.com/ElrondNetwork/elrond-go/crypto/signing/mcl/singlesig"
+)
+
+func main(){
+	args := os.Args
+
+	if len(os.Args) != 3 {
+	    fmt.Printf("invalid number of arguments: got %d expected 2 \n", len(os.Args)-1)
+	    return
+	}
+
+	singleSigner := singlesig.BlsSingleSigner{}
+
+	messageToSign, err := hex.DecodeString(args[1])
+	if err != nil {
+		fmt.Println(err.Error())
```

Comment:
Also, `os.Exit(something other than 0)` as well? So that `erdpy` (subprocess call) will detect eventual errors.

See example at: https://github.com/ElrondNetwork/arwen-wasm-vm/blob/master/cmd/arwen/main.go.

Reply:
done refactored.